media/heif/bmff/bmff.go

```diff
@@ -699,39 +699,163 @@ func parseItemLocationBox(outer *box, br *bufReader) (Box, error) {
 	if err != nil {
 		return nil, err
 	}
+	if fb.Version > 2 {
+		return nil, fmt.Errorf("go4 heif: error unsupported iloc version of %v", fb.Version)
+	}
 	ilb := &ItemLocationBox{
 		FullBox: fb,
 	}
-	buf, err := br.Peek(4)
+	buf, err := br.Peek(2)
 	if err != nil {
 		return nil, err
 	}
 	ilb.offsetSize = buf[0] >> 4
 	ilb.lengthSize = buf[0] & 15
 	ilb.baseOffsetSize = buf[1] >> 4
-	if fb.Version > 0 { // version 1
+	switch fb.Version {
+	case 0:
+		ilb.indexSize = 0
+	case 1, 2:
 		ilb.indexSize = buf[1] & 15
+	default:
+		return nil, fmt.Errorf("go4 heif: error unsupported iloc version of %v", fb.Version)
```

Comment:
Other `Errorf`s in this package are simplye `fmt.Errorf("error message: context value")`:
```suggestion
		return nil, fmt.Errorf("unsupported iloc version: %d", fb.Version)
```

Additionally, by using `%d` it allows `go vet` to have the context that this should be a numeric type, and it can throw an error when a string or struct is used instead.

Reply:
Changed to using %d.